posts/2020-09-29-microprofile-openapi-beta-200011.adoc

```diff
@@ -0,0 +1,253 @@
+
+[source, java]
+----
+@Schema(
+    name = "User",
+        description = "Telephone number to contact the user”,
+        implementation = User.class,
+        properties = {
+            @SchemaProperty(
+                name = "phone",
+                description = "Telephone number to contact the user”
+            )
+        }
+    )} 
+----
+[#requestbodyschema]
+==== @RequestBodySchema
+
+The @RequestBodySchema annotation has been added to provide a shorthand mechanism to specify the schema for a request body. Previously, defining the schema for a request body required the use of three different annotations, as follows:
+[source, java]
+----
+@RequestBody(
+    content = {
+        @Content(
+            schema = @Schema(
+                implementation = MyRequestObject.class
+            )
+        )
+    }
+)
+----
+
+With the introduction of the @RequestBodySchema annotation, this can be simplified to the following:
+
+[source, java]
+----
+@RequestBodySchema(MyRequestObject.class) 
+----
+
+[#apiresponseschema]
+==== @APIResponseSchema
+
+The @APIResponseSchema annotation has been added to provide a shorthand mechanism to specify the schema for a response body. Previously, defining the schema for a response body required the use of three different annotations, as follows:
+
+[source, java]
+----
+@APIResponse(
+    content = {
+            @Content(
+                schema = @Schema(
+                    implementation = MyResponseObject.class
+                )
+            )
+        }
+    )
+----
+
+With the introduction of the @RequestBodySchema annotation, this can be simplified to the following:
+
+[source, java]
+----
+@APIResponseSchema(MyResponseObject.class)
+----
+
+[#mpopenapischeme]
+==== mp.openapi.schema.*
+The mp.openapi.schema.* MicroProfile Config property prefix has been added to allow you to define the schema for Java classes using configuration rather than code. The use of this property is functionally equivalent to the use of the @Schema annotation on a Java class, but may be used in cases where you do not have access to the source code of a class. The remainder of the property key must be the fully-qualified class name and the value must be a valid OpenAPI schema object, specified in the JSON format.
+For example, in the case where an application needs to represent dates in epoch milliseconds, the following configuration could be used (line escapes and indentation added for readability):
+
+[source, java]
+----
+mp.openapi.schema.java.util.Date = { \
+   "name": "EpochMillis" \
+   "type": "number", \
+   "format": "int64", \
+   "description": "Milliseconds since January 1, 1970, 00:00:00 GMT" \
+ }
+----
+
+Enable OpenAPI 2.0 in the server.xml, along with any other features you’re using.
+
+[source, xml]
+----
+<featureManager>
+       <feature>mpOpenAPI-2.0</feature>
+   </featureManager>
```

Comment:
The closing XML tag needs to line up with the opening XML tag.
And make sure any XML indentation of the `feature` element is consistent across examples throughout the blog post.

Reply:
Done